src/modules/DataStore/BadgeCase.ts

```diff
@@ -40,15 +40,16 @@ export default class BadgeCase implements Feature {
         }
 
         json.forEach((hasBadge, index) => {
-            this.badgeList[index] = hasBadge;
+            this.badgeList[index](hasBadge);
         });
     }
 
     toJSON(): Record<string, any> {
         let shouldReturn = false;
         // We only want to save upto the highest badge we have obtained,
         // everything else is assumed to be false
-        return [...this.badgeList]
+        return this.badgeList
+            .map((v) => v())
```

Comment:
alternative that doesn't require making an anonymous function:
```suggestion
            .map(ko.unwrap)
````
I am not sure I like this suggestion any more though. Such a simple operation as function application I had hoped would have a native javascript thing we could map, but I couldn't find anything that did what I wanted on Function or Function.prototype

Reply:
I've made this change for BadgeCase, Farming, Shards, and Wallet. Its not a huge improvement for characters or readability, but `ko.unwrap` will guard against unwrapping an observable or primitive, so its definitely safer